midterm-2023-11-07/code/5263-angela-kostadinova/script.ts

```diff
@@ -0,0 +1,106 @@
+interface Movie {
+
+id: number;
+title: string;
+director: string;
+year: number;
+genre: string;
+plot: string;
+cast: string[];
+oscars: number;
+}
+async function LoadingTheData() {
+    const dataUri = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-07/data/movies.json";
+    const response = await fetch(dataUri);
+
+    if (!response.ok) {
+        throw new Error("Ooops...Sorry, I cannot take requested data!");
+    }
+
+    const data = await response.json();
+    return data;
+}
+document.addEventListener("DOMContentLoaded", siteCode);
+function CastListGenerate(cast: string[]) {
+    const ul = document.createElement("ul");
+    for (const actor of cast) {
+        const li = document.createElement("li");
+        li.innerHTML = actor;
+        ul.appendChild(li);
+    }
+    return ul;
+}
+function DisplayAllMovies(movies: Movie[]) {
+    const container = document.getElementById("movie-container")!;
+    container.innerHTML = "";
+
+    for (const movie of movies) {
+        const movieTable = generateMovieTable(movie);
+        container.appendChild(movieTable);
+    }
+}
+function generateMovieTable(movie: Movie) {
+    const table = document.createElement("div");
+    table.classList.add("movie-table");
+    const movieData = [
```

Comment:
Generating elements and displaying movie data +15
Using an array and loop-adding it +5 extra